.gitignore

```diff
@@ -0,0 +1,4 @@
+
+# this is bunch of pyc files "python compiled" which are generated whenever you run python code.  They should not be included in a repo.
+
+__pycache__/
```

Comment:
So this removes / ignores the cached Python files?

Reply:
Yep, anything in your [.gitignore](https://www.atlassian.com/git/tutorials/saving-changes/gitignore) file is ignored during `git add`, `git commit`, `git push` operations.  We typically ignore "compiled versions of the code as they can be regenerated quite easily by whoever pulls down the repo.  

Also, a common pattern is to have a virtual environment for python (you'll see `venv/bin/activate` around on the web) on your machine.  This virtual environment folder is typically excluded as well.

Finally, for development purposes, you will sometimes want to keep an environmental variable, you can keep track of those environmental variables with a file like `dev.env` and `prod.env`, but put `*.env` in your .gitignore file so you never commit secret variable into the codebase.

Reply:
Yep, anything in your [.gitignore](https://www.atlassian.com/git/tutorials/saving-changes/gitignore) file is ignored during `git add`, `git commit`, `git push` operations.  We typically ignore "compiled versions of the code as they can be regenerated quite easily by whoever pulls down the repo.  

Also, a common pattern is to have a virtual environment for python (you'll see `venv/bin/activate` around on the web) on your machine.  This virtual environment folder is typically excluded as well.

Finally, for development purposes, you will sometimes want to keep an environmental variable, you can keep track of those environmental variables with a file like `dev.env` and `prod.env`, but put `*.env` in your .gitignore file so you never commit secret variable into the codebase.